test/e2e/external_gateways.go

```diff
@@ -1008,17 +1095,50 @@ EOF
 				framework.ExpectNoError(err, "failed to setup FRR peer %s in %s", a, containerName)
 			}
 		}
-		cmd := []string{"docker", "exec", containerName, "/usr/lib/frr/frrinit.sh", "start"}
+		cmd := []string{containerRuntime, "exec", containerName, "/usr/lib/frr/frrinit.sh", "start"}
 		_, err := runCommand(cmd...)
 		framework.ExpectNoError(err, "failed to start frr in %s", containerName)
 	}
 }
 
 func isBFDPaired(container, peer string) bool {
-	res, err := runCommand("docker", "exec", container, "bash", "-c", fmt.Sprintf("vtysh -c \"show bfd peer %s\"", peer))
+	res, err := runCommand(containerRuntime, "exec", container, "bash", "-c", fmt.Sprintf("vtysh -c \"show bfd peer %s\"", peer))
 	framework.ExpectNoError(err, "failed to check bfd status in %s", container)
 	if strings.Contains(res, "Status: up") {
 		return true
 	}
 	return false
 }
+
+// When running on host network we clean the routes and ips we added previously
+func cleanRoutesAndIPs(containerName string, addressesv4, addressesv6 gatewayTestIPs) {
+	testIPv6 := addressesv6.srcPodIP != "" && addressesv6.nodeIP != ""
+	testIPv4 := addressesv4.srcPodIP != "" && addressesv4.nodeIP != ""
+
+	if testIPv4 {
+		cleanRoutesAndIPsForFamily(containerName, 4, addressesv4)
+	}
+	if testIPv6 {
+		cleanRoutesAndIPsForFamily(containerName, 6, addressesv6)
+	}
+}
+
+func cleanRoutesAndIPsForFamily(containerName string, family int, addresses gatewayTestIPs) {
+	var err error
+	var addressMask string
+	addressSelector := fmt.Sprintf("-%d", family)
+	mask := 32
+	if family == 6 {
+		mask = 128
+	}
+	ginkgo.By(fmt.Sprintf("Removing the destination ips from %s (IPv%d)", containerName, family))
+	for _, address := range addresses.targetIPs {
+		addressMask = fmt.Sprintf("%s/%d", address, mask)
+		_, err = runCommand(containerRuntime, "exec", containerName, "ip", addressSelector, "address", "del", addressMask, "dev", "lo")
+		framework.ExpectNoError(err, "failed to del the loopback ip to dev lo on the test container %s", containerName)
```

Comment:
will be useful to include addressMask to the error so we know which address family had the issue

Reply:
added